cmd/proxygenerator/interceptor.go

```diff
@@ -500,7 +674,14 @@ func walk(desired []types.Type, typ types.Type, records *map[string]*TypeRecord,
 		// All the Get... methods return the relevant protobuf as the first result
 		resultType := sig.Results().At(0).Type()
 
-		if checkDirectPayload && resultType.String() == "*go.temporal.io/api/common/v1.Payload" {
+		hasDirectMatch := false
+		for _, directMatchType := range directMatchTypes {
+			if resultType.String() == types.NewPointer(directMatchType).String() {
+				hasDirectMatch = true
```

Comment:
Don't love that I had to do this but I could not for the life of me get reflection to be equal properly